dot/sync/chain_sync.go

```diff
@@ -941,37 +941,27 @@ func (cs *chainSync) getHighestBlock() (int64, error) {
 	ps := cs.peerState
 	cs.RUnlock()
 
-	if len(ps) <= 0 {
+	if len(ps) == 0 {
 		return 0, errNoPeers
 	}
 
-	// improve performance returning first highestBlock
-	if len(ps) == 1 {
-		for _, p := range ps {
-			if p.number == nil {
-				return 0, errNilBlockData
-			}
-			return p.number.Int64(), nil
-		}
-	}
+	highestBlock := big.NewInt(-1)
 
-	numbers := []*big.Int{}
 	for _, ps := range ps {
 		if ps.number == nil {
 			continue
 		}
-		numbers = append(numbers, ps.number)
+		if ps.number.Cmp(highestBlock) < 0 { // ps.number < highestBlock: https://pkg.go.dev/math/big#Int.Cmp
```

Comment:
nit unneeded comment, one can just check the Cmp comment. Also this will be changed soon to use `uint` instead of `*big.Int` (in a few weeks) :wink: 

```suggestion
		if ps.number.Cmp(highestBlock) < 0 {
```

Reply:
Done in c2e3d69